Mage/src/main/java/mage/abilities/common/CaseAbility.java

```diff
@@ -0,0 +1,128 @@
+import mage.util.CardUtil;
+
+/**
+ * The Case mechanic was added in Murders at Karlov Manor [MKM].
+ * <ul>
+ * <li>Each Case has two special keyword abilities: to solve and solved.</li>
+ * <li>"To Solve — [condition]" means "At the beginning of your end step,
+ * if [condition] and this Case is not solved, it becomes solved."</li>
+ * <li>The meaning of "solved" differs based on what type of ability follows it.
+ * "Solved — [activated ability]" means "[Activated ability].
+ * Activate only if this Case is solved." Activated abilities contain a colon.
+ * They're generally written "[Cost]: [Effect]."</li>
+ * <li>"Solved — [Triggered ability]" means "[Triggered ability].
+ * This ability triggers only if this Case is solved."
+ * Triggered abilities use the word "when," "whenever," or "at."
+ * They're often written as "[Trigger condition], [effect]."</li>
+ * <li>"Solved — [static ability]" means "As long as this Case is solved, [static ability]."
+ * Static abilities are written as statements, such as "Creatures you control get +1/+1"
+ * or "Instant and sorcery spells you cast cost {1} less to cast."</li>
+ * <li>"To solve" abilities will check for their condition twice:
+ * once when the ability would trigger, and once when it resolves.
+ * If the condition isn't true at the beginning of your end step,
+ * the ability won't trigger at all.
+ * If the condition isn't true when the ability resolves, the Case won't become solved.</li>
+ * <li>Once a Case becomes solved, it stays solved until it leaves the battlefield.</li>
+ * <li>Cases don't lose their other abilities when they become solved.</li>
+ * <li>Being solved is not part of a permanent's copiable values.
+ * A permanent that becomes a copy of a solved Case is not solved.
+ * A solved Case that somehow becomes a copy of a different Case stays solved.</li>
+ * </ul>
+ *
+ * @author DominionSpy
+ */
+public class CaseAbility extends SimpleStaticAbility {
+
+    public CaseAbility(Ability initialAbility, Condition toSolveCondition, Ability solvedAbility) {
+        super(Zone.ALL, null);
+
+        addSubAbility(initialAbility.withNameReplacement("this Case"));
+
+        addSubAbility(new CaseSolveAbility(toSolveCondition));
+
+        addSubAbility(solvedAbility.withNameReplacement("this Case")
+                .withFlavorWord("Solved"));
+    }
+
+    protected CaseAbility(final CaseAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public CaseAbility copy() {
+        return new CaseAbility(this);
+    }
+
+    @Override
+    public String getRule() {
+        return super.getRule().replace("{this}", "this Case");
+    }
+}
+
+class CaseSolveAbility extends BeginningOfEndStepTriggeredAbility {
+
+    CaseSolveAbility(Condition condition) {
+        super(new SolveEffect(), TargetController.YOU,
+                new CompoundCondition(condition, SolvedSourceCondition.UNSOLVED), false);
+        withFlavorWord("To solve &mdash; ");
```

Comment:
In my opinion it's totally fine that this text is getting italicized. However, there's already an em dash so you don't need to add it here.

![image](https://github.com/magefree/mage/assets/12538125/ddc2171e-bec1-4390-a314-a435f3459de4)